src/database/mod.rs

```diff
@@ -537,6 +549,22 @@ impl Database {
         }
     }
 
+    pub async fn update_latest_insertion_timestamp(
+        &self,
+        insertion_timestamp: DateTime<Utc>,
+    ) -> Result<(), Error> {
+        let query = sqlx::query(
+            r#"
+            INSERT INTO latest_insertion_timestamp (insertion_timestamp)
+            VALUES ($1)
+            "#,
```

Comment:
This will add a new entry into the table every time this function is called instead of updating the existing entry. 

To update the existing entry, you could an "id" field and make sure to update that specific id each time. Alternatively, you could add a constraint similar to the latest deletion table.

Reply:
noted, I constrained it similarly like with the deletion root